api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -134,15 +134,16 @@ private boolean isDuplicate(QueueEntry queueEntry) {
 	public QueueEntry transitionQueueEntry(QueueEntryTransition queueEntryTransition) {
 		// Create a new queue entry
 		QueueEntry queueEntryToStart = queueEntryTransition.constructNewQueueEntry();
-		if (isDuplicate(queueEntryToStart)) {
-			throw new DuplicateQueueEntryException("queue.entry.duplicate.patient");
-		}
 		
 		// End the initial queue entry
 		QueueEntry queueEntryToStop = queueEntryTransition.getQueueEntryToTransition();
 		queueEntryToStop.setEndedAt(queueEntryTransition.getTransitionDate());
 		getProxiedQueueEntryService().saveQueueEntry(queueEntryToStop);
 		
+		if (isDuplicate(queueEntryToStart)) {
+			throw new DuplicateQueueEntryException("queue.entry.duplicate.patient");
+		}
+		
```

Comment:
Do we have tests/can we confirm that throwing this exception causes the "saveQueueEntry" listed above to be rolled back (which I think was what this fix was trying to achieve).

I would assumed it would be if the method is transactionally, but would be good to verify (which there is likely already a test for, just wanted to confirm).

Alternatively, could the isDuplicate check be changed to be smart enough to ignore the queueEntryToStop when checking for duplicates?